### ISSUE TITLE
Add ability to search for episodes before airdate

### ISSUE THREAD

**Issue by andyj682** [SiCKRAGETV/sickrage-issues#622]
Occasionally, networks release new episodes of shows on demand as early as a week before the listed airdate. These episodes often become available for download as soon as they show up on demand. However, SickRage doesn't seem to search for the next upcoming episode until the actual airdate. This makes sense, since most of the time searching for episodes early will just yield spam or mislabeled episodes, but it's also annoying to not be able to snatch episodes that become available early right away when they're legit.

So, first of all: Is my understanding correct that SickRage isn't even supposed to search for new episodes until they've aired? If so, it'd be great to add an option for each show to "search for new episodes before expected airdate" (or something like that). I think CouchPotato has an option like this, for example. And if I'm wrong and SickRage is supposed to be searching for new episodes far in advance of their airdate, it's not working for me on at least one specific show even after removing/readding. 

Again, exact experience for me on this show I have in mind is: Next new episode is always available about a week before airdate, but SickRage won't ever find it through daily searches. I can search for the episode manually and it'll download just fine--so issue is definitely that SickRage isn't searching for it automatically.

**Comment by coolius** [comment]
I would like to second this request. The show "Forever (2014)" is always available a day earlier than the date quoted on TVRage/TVDB as they use the US airdate despite the fact that the show is broadcast in Canada (and therefore available to download) 24 hours before.

I had envisioned this being a per-show option, with the user able to choose the number of days (1-7?) before the stated airdate SickRage should begin searching.

Many thanks.

**Comment by neoatomic** [comment]
This is more a problem of not correct air-times on the indexer. And it should be corrected there.
Currently i dont see a possibility to add this...

**Comment by andyj682** [comment]
Fair enough if this won't be addressed--it's low priority, for sure--but it's actually not an issue with air dates on the indexer. We're talking about shows where the episodes, for whatever reason, are sometimes available several days before the official (and correct) air date. Clearly there's something in the current code that tells SickRage NOT to search for episodes if they haven't yet aired, so there would probably be a way to let it lift that restriction on a show-by-show basis, right? Again, not familiar enough with the script to know how hard it would be to do this, and it could definitely be more trouble than it's worth, but it's a legit issue on certain shows and it's pretty easy to see the "possibility" to add the feature.

**Comment by neoatomic** [comment]
@andyj682 Its pretty simple, when a show hasn't aired it cant be downloaded. If the show is out before the air-date on the indexer than the air-date is wrong.
(It rarely happens that an episode leaks before the air-date, but you never know that in advance.)
Searching before that time will only snatch fakes and hammer users APIs with unnecessary requests. In short not a goo idea...

**Comment by andyj682** [comment]
@neoatomic I'm aware of how this stuff works; I promise. The specific example I was thinking of when I made the request was "Episodes" on Showtime. That show airs on Sunday nights--that is the official and correct air date you'd find in any TV database. But Showtime made every episode this season available "on demand" 5-6 days ahead of the official air date, which means the episodes were legitimately available several days before the correct air date. This happens with a few other Showtime shows, too. Not sure about other networks. Is it a lot of shows? No, probably only enough to count on two hands, and none of the biggies. Is it a legitimate issue, though? Yes, because if you happen to watch one of those shows, having to remember to search for episodes manually every week kind of defeats the purpose of a program like SickRage. 

I agree with you that it'd be dumb to make searching for new episodes before the expected air date a global option. That's why I suggested a per-show option, which users would obviously only select if it's a show that's consistently available before the air date. CouchPotato has this sort of functionality, albeit not on a per-movie basis.
